pre_commit_hooks/trailing_whitespace_fixer.py

```diff
@@ -21,10 +37,19 @@ def fix_trailing_whitespace(argv=None):
         'grep', '-l', '[[:space:]]$', *args.filenames, retcode=None
     ).strip().splitlines()
 
+    # combine all extension arguments, splitting at ',' and normalizing them
+    # (lowercase and remove unnecessary leading '.' which may be present)
+    md_args = vars(args).get('markdown_linebreak_ext', ['md', 'markdown'])
+    md_exts = [x.lower().lstrip('.') for x in ','.join(md_args).split(',')]
+    all_markdown = '*' in md_exts
+
     if bad_whitespace_files:
         for bad_whitespace_file in bad_whitespace_files:
+            # get extension ([1]); remove leading '.' ([1:]); and make lowercase
+            extension = os.path.splitext(bad_whitespace_file)[1][1:].lower()
```

Comment:
This should be enough to simplify it a bit:

```diff
-    md_exts = [x.lower().lstrip('.') for x in ','.join(md_args).split(',')]
+    md_exts = ['.' + x.lower().lstrip('.') for x in ','.join(md_args).split(',')]
```

The check for contained dots then becomes a check for `.count(2)`